packages/neon_framework/packages/account_repository/lib/src/account_repository.dart

```diff
@@ -0,0 +1,329 @@
+  final AccountStorage _storage;
+
+  (NextcloudClient, core.ClientFlowLoginV2PollRequestApplicationJson)? _flow;
+
+  final BehaviorSubject<({String? active, BuiltMap<String, Account> accounts})> _accounts =
+      BehaviorSubject.seeded((active: null, accounts: BuiltMap()));
+
+  /// A stream of account information.
+  ///
+  /// The initial state of the stream will be `(active: null, accounts: BuiltMap())`.
+  Stream<({String? active, BuiltList<Account> accounts})> get accounts => _accounts.stream.map((e) {
+        final accounts = e.accounts.values.toBuiltList();
+
+        return (active: e.active, accounts: accounts);
+      }).asBroadcastStream();
+
+  /// Whether accounts are logged in.
+  bool get hasAccounts {
+    return _accounts.hasValue && _accounts.value.accounts.isNotEmpty;
+  }
+
+  /// Retrieves the account associated with the given [accountID].
+  Account? accountByID(String accountID) {
+    return _accounts.value.accounts[accountID];
+  }
+
+  /// Checks whether the provided [url] is a valid login qr code.
+  static bool isLogInQRCode({required String url}) {
+    return LoginQRcode.tryParse(url) != null;
+  }
+
+  /// Loads the stored account information.
+  ///
+  /// If specified [rememberLastUsedAccount] takes precedence over [initialAccount].
+  /// If the stored or provided initial account are not in the list of stored credentials
+  /// they are silently ignored and the default of `first account in the list` is used.
+  Future<void> loadAccounts({
+    String? initialAccount,
+    bool rememberLastUsedAccount = false,
+  }) async {
+    final stored = await _storage.readAccounts();
+    final accounts = BuiltMap<String, Account>.build((b) {
+      for (final credentials in stored) {
+        b[credentials.id] = Account((b) {
+          b
+            ..credentials.replace(credentials)
+            ..client = buildClient(
+              httpClient: _httpClient,
+              userAgent: _userAgent,
+              credentials: credentials,
+            );
+        });
+      }
+    });
+
+    var active = initialAccount;
+    if (rememberLastUsedAccount) {
+      active = await _storage.readLastAccount();
+    }
+
+    if (active == null || !accounts.containsKey(active)) {
+      active = accounts.keys.firstOrNull;
+    }
+
+    _accounts.add(
+      (
+        active: active,
+        accounts: accounts,
+      ),
+    );
+  }
+
+  // TODO: this doesn't really belong here.
+  /// Fetches the status of the server.
+  ///
+  /// May throw a [FetchStatusFailure].
+  Future<core.Status> getServerStatus(Uri serverURL) async {
+    final client = buildUnAuthenticatedClient(
+      httpClient: _httpClient,
+      userAgent: _userAgent,
+      serverURL: serverURL,
+    );
+
+    try {
+      final response = await client.authentication.core.getStatus();
+      return response.body;
+    } on http.ClientException catch (error, stackTrace) {
+      Error.throwWithStackTrace(FetchStatusFailure(error), stackTrace);
+    }
+  }
```

Comment:
This is still in the account_repository.
We can still refactor it later on, and I'm not sure yet about how to best manage the NextcloudClients for it.

Once authenticated, the client of the account should be used everywhere, but here we explicitly need to work without any authentication.

Reply:
I agree and I have a wip branch where I implement this.
The issue is that we want to give it `core.$Client` as a constructor parameter for easier mocking.
The core capabilities require an authenticated user while this status check needs to be done before we are authenticated.

This means we would need to insert the `server_repository` with an unauthenticated `core.$Client` (and throw when one tries to access other methods).
Once the user is logged in we would need to rebuild that repository with an authenticated client.
When a user then tries to log in to a different server we would again need a different client.

I just think that this is to much state specific information in the repositroy and therefore leave it in here for now.
It isn't that much code duplication if we also add the same method to the `server_repository`.

Reply:
Capabilities doesn't need to be authenticated. It is only possible that it will return different data when logged in vs. not.

Reply:
But we probably rely on some authenticated capability, so it doesn't matter.